commands/uploader.go

```diff
@@ -327,7 +329,11 @@ func (c *uploadContext) Await() {
 	c.trackedLocksMu.Unlock()
 
 	if avoidPush {
-		Error("WARNING: The above files would have halted this push.")
+		if c.lockVerifyState == verifyStateEnabled {
+			Exit("ERROR: Cannot update locked files.")
+		} else {
+			Error("WARNING: The above files would have halted this push.")
```

Comment:
Nit: this could be de-dented since `Exit()` halts execution. Not sure what your preference is here, since this isn't as obvious as a `return` or `panic()` instead of a conditional.

More generally speaking, I'm glad that this warning is still being kept. I originally added it to warn about locks that _would_ have halted the push had lock verification been binding. This is rad, since if the state of lock verification is undefined, we can pretend like it is and _still_ warn about illegally modified files.

That being said, the behavior of `verifyLocks()` is to make an early-return if the lock verification state is disabled. That would cause this message not to be printed at all (even if the server _did_ support the locking API _and_ there were illegally modified files in the push). I think we should change the behavior of `verifyLocks()` to _always_ try and make the API call, and only propagate the error if one occurs and the state is enabled or undefined.

That would allow us to always present this message even if lock verification is turned off. I think this hardens the requirement of owning your locked files and would be a positive improvement 👍 .